include/crocoddyl/core/actions/lqr.hxx

```diff
@@ -0,0 +1,194 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2018-2020, LAAS-CNRS, University of Edinburgh
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#include "crocoddyl/core/utils/exception.hpp"
+
+namespace crocoddyl {
+
+template <typename Scalar>
+ActionModelLQRTpl<Scalar>::ActionModelLQRTpl(const std::size_t& nx, const std::size_t& nu, bool drift_free)
+    : Base(boost::make_shared<StateVector>(nx), nu, 0), drift_free_(drift_free) {
+  // TODO(cmastalli): substitute by random (vectors) and random-orthogonal (matrices)
+  Fx_ = MatrixXs::Identity(nx, nx);
+  Fu_ = MatrixXs::Identity(nx, nu);
+  f0_ = VectorXs::Ones(nx);
+  Lxx_ = MatrixXs::Identity(nx, nx);
+  Lxu_ = MatrixXs::Identity(nx, nu);
+  Luu_ = MatrixXs::Identity(nu, nu);
+  lx_ = VectorXs::Ones(nx);
+  lu_ = VectorXs::Ones(nu);
+}
+
+template <typename Scalar>
+ActionModelLQRTpl<Scalar>::~ActionModelLQRTpl() {}
+
+template <typename Scalar>
+void ActionModelLQRTpl<Scalar>::calc(const boost::shared_ptr<ActionDataAbstract>& data,
+                                     const Eigen::Ref<const VectorXs>& x, const Eigen::Ref<const VectorXs>& u) {
+  if (static_cast<std::size_t>(x.size()) != state_->get_nx()) {
+    throw_pretty("Invalid argument: "
+                 << "x has wrong dimension (it should be " + std::to_string(state_->get_nx()) + ")");
+  }
+  if (static_cast<std::size_t>(u.size()) != nu_) {
+    throw_pretty("Invalid argument: "
+                 << "u has wrong dimension (it should be " + std::to_string(nu_) + ")");
+  }
+
+  if (drift_free_) {
+    data->xnext = Fx_ * x + Fu_ * u;
```

Comment:
Missing noalias()

Reply:
Thanks!
(I didn't look at the implementation in this PR)